### PR TITLE
raspberrypi-wireless-firmware, raspberrypi-firmware, linux-rpi: sync to Raspberry Pi OS Bookworm release

### PR DESCRIPTION
**Pull request by peat-psuwit**
## Description of changes

This PR bumps versions of Raspberry Pi firmwares & kernel to match the version used by Raspberry Pi OS Bookworm release. This includes:

- raspberrypi-wireless-firmware: unstable-2023-05-04 -> unstable-2023-11-15, which combines:
  * bluez-firmware 1.2-9+rpt2
  * firmware-nonfree 1:20230210-5+rpt2
- raspberrypi-firmware: 1.20230405 -> stable_20231123
- linux-rpi: 6.1.21-1.20230405 -> 6.1.63-stable_20231123

There's a minor (potentially breaking) change in raspberrypi-wireless-firmware, where following files. which are actually part of upstream bluez-firmware and not RPi's firmwares, are no longer included in this package:

- BCM2033-\*.bin, BCM2033-\*.hex
- BCM-LEGAL.txt
- bcm_changeba
- bcm_changeid
- bcm_hex2bin

## Things done

- Built on platform(s)
  - [x] x86_64-linux (only firmware packages)
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - Tested by installing this PR into my Raspberry Pi 4.
  - The built kernel boots. Tested functionalities are HW media decoding and Wi-Fi.
  - Bluetooth is not tested, as my RPi 4 doesn't have Bluetooth stack configured.
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peat-psuwit** [comment]
Ping the developers.

File: pkgs/os-specific/linux/kernel/linux-rpi.nix [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 let
   # NOTE: raspberrypifw & raspberryPiWirelessFirmware should be updated with this
-  modDirVersion = "6.1.21";
-  tag = "1.20230405";
+  modDirVersion = "6.1.58";
+  tag = "stable_20231030";
```
Comment (dasJ) [inline pkgs/os-specific/linux/kernel/linux-rpi.nix:6]: Unfortunately, `stable_20231123` is out now :/

**Comment by dasJ** [comment]
Shouldn't this also get backported to 23.11?

File: pkgs/os-specific/linux/kernel/linux-rpi.nix (hunk 1 above)
Comment (peat-psuwit) [reply pkgs/os-specific/linux/kernel/linux-rpi.nix]: Updated to stable_20231123 for both firmwares and kernel.

**Comment by peat-psuwit** [comment]
> Shouldn't this also get backported to 23.11?

Quite possibly. However, I guess depending on how raspberrypi-firmware is used, it might not be straight forward.

**Comment by leo60228** [comment]
Would it make sense to add `linux_rpi5` here too? See #260754: doing so would already technically work with `boot.loader.raspberryPi`, should allow a mostly-supported setup once there are usable U-Boot builds for the Pi 5, and might be able to be used with EDK2.

**Comment by gador** [comment]
Since this update, neiher hydra nor I am able to build `linux-6.1.63-stable_20231123`
See https://hydra.nixos.org/build/246720053

**Comment by peat-psuwit** [comment]
Right... `module_pci_driver()` is a macro defined in `linux/pci.h`, so somehow that macro is not defined in the build. I think this is related to the fact that this warning appears:

```
WARNING: unmet direct dependencies detected for MFD_RP1
  Depends on [n]: HAS_IOMEM [=y] && PCI [=n]
  Selected by [m]:
  - DRM_RP1_DSI [=m] && HAS_IOMEM [=y] && DRM [=m]
  - DRM_RP1_DPI [=m] && HAS_IOMEM [=y] && DRM [=m]
  - DRM_RP1_VEC [=m] && HAS_IOMEM [=y] && DRM [=m]
```

For context, RP1 is a new chipset introduced in Raspberry Pi 5 and is connected via PCI Express bus. Given that Raspberry Pi 3 doesn't even have a PCI Express bus, I think `MFD_RP1` shouldn't be enabled at all.

Will investigate.

**Comment by peat-psuwit** [comment]
Please see https://github.com/NixOS/nixpkgs/pull/284035

**Pull request merged**